packages/Tabs/src/Tabs.js

```diff
@@ -26,73 +25,96 @@ export default function Tabs(props) {
-  const [currentFocusIndex, setFocusIndex] = React.useState(indexToUse);
-  let tabListRef = React.useRef(null);
+  const [focusIndex, setFocusIndex] = React.useState(indexToUse || 0);
+  const refList = React.useRef(null);
+  const [idTabs] = React.useState(uuidv4());
 
-  React.useEffect(() => {
+  React.useLayoutEffect(() => {
     setActiveIndex(indexToUse);
-  }, [indexToUse, setActiveIndex]);
+  }, [indexToUse]);
+
+  React.useLayoutEffect(() => {
+    const items = getItems(refList);
+    if (indexToUse !== null && items.length > 0 && isItemDisabled(items[indexToUse])) {
+      const itemIndexes = getItemIndexes(refList, { hasDisabledItems: false });
 
-  function focusAndSetIndex(index) {
-    tabListRef.querySelectorAll("[data-pka-anchor='tab'], [data-pka-anchor='tab-link']")[index].focus();
+      if (itemIndexes.length > 0) {
+        setActiveIndex(itemIndexes[0]);
+      }
+    }
+  }, []);
+
+  function focusItem(index) {
+    getItems(refList)[index].focus();
     setFocusIndex(index);
   }
 
-  const setTabListRef = ref => {
-    tabListRef = ref;
-  };
-
   const handleClickTab = (event, index) => {
     event.preventDefault();
 
-    if (onClickTab) {
+    if (isControlled) {
       onClickTab(index);
     } else {
       setActiveIndex(index);
     }
+    setFocusIndex(index);
   };
 
-  // TODO: Disabled tab items should also get focus on keyboard interaction
-  // https://github.com/acl-services/paprika/issues/310
-  const onKeyDown = (event, currentIndex) => {
-    const { "Tabs.List": TabsList } = extractChildren(children, ["Tabs.List"]);
-    const tabs = TabsList.props.children.filter(child => child !== null);
-    const enabledIndexes = tabs
-      .map((tab, index) => (tab.props.isDisabled === true ? null : index))
-      .filter(index => index !== null);
-    const enabledSelectedIndex = enabledIndexes.indexOf(currentIndex);
-    const count = enabledIndexes.length;
-
-    if (nextKeys.includes(event.key)) {
-      const nextEnabledIndex = (enabledSelectedIndex + 1) % count;
-      const nextIndex = enabledIndexes[nextEnabledIndex];
-      focusAndSetIndex(nextIndex);
-    } else if (prevKeys.includes(event.key)) {
-      const nextEnabledIndex = (enabledSelectedIndex - 1 + count) % count;
-      const nextIndex = enabledIndexes[nextEnabledIndex];
-      focusAndSetIndex(nextIndex);
-    } else if (event.key === "Home") {
-      focusAndSetIndex(enabledIndexes[0]);
-    } else if (event.key === "End") {
-      focusAndSetIndex(enabledIndexes[count - 1]);
+  const handleKeyDown = event => {
+    if (Object.values(keyTypes).includes(event.key)) {
+      event.stopPropagation();
+
+      const itemIndexes = getItemIndexes(refList);
+      const enabledSelectedIndex = itemIndexes.indexOf(focusIndex);
+      const count = itemIndexes.length;
+
+      switch (event.key) {
+        case keyTypes.RIGHT:
+        case keyTypes.DOWN:
+          focusItem(itemIndexes[(enabledSelectedIndex + 1) % count]);
```

Comment:
this will loop over the tabs? most HTML inputs don't loop over when using the keyboard any reason for this one to do it?

Reply:
This is the behaviour specified in the WAI ARIA Authoring Practices:
> Right Arrow: Moves focus to the next tab. If focus is on the last tab element, moves focus to the first tab. 

Source: https://www.w3.org/TR/wai-aria-practices-1.1/#keyboard-interaction-19

I think this pattern makes more sense for a tabs control because the number of tabs is relatively small.